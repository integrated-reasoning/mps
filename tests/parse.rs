mod tests {
  use color_eyre::Result;
  use num_traits::float::Float;
  cfg_if::cfg_if! {
    if #[cfg(feature = "trace")] {
      use nom_locate::LocatedSpan;
      use nom_tracable::TracableInfo;
    }
  }

  fn parse<T: Float>(input: &'static str) -> Result<mps::Parser<'_, f32>> {
    cfg_if::cfg_if! {
      if #[cfg(feature = "trace")] {
        let info = TracableInfo::new().forward(false).backward(false);
        let (_, parsed) = mps::Parser::<T>::parse(LocatedSpan::new_extra(input, info))?;
      } else {
        let (_, parsed) = mps::Parser::<T>::parse(&input)?;
      }
    }
    Ok(parsed)
  }

  #[test]
  fn test_parse_agg() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/agg"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_ship04l() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/ship04l"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_d2q06c() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/d2q06c"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_e226() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/e226"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_nl25fv47() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/25fv47"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_bore3d() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/bore3d"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_ganges() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/ganges"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_adlittle() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/adlittle"
    ))?);
    Ok(())
  }

  #[ignore] // TODO: Fix (fails in row_line and columns)
  fn _test_parse_forplan() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/forplan"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sc205() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sc205"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_nl80bau3b() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/80bau3b"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scrs8() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scrs8"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_wood1p() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/wood1p"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_boeing1() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/boeing1"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_kb2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/kb2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_ship08s() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/ship08s"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scfxm1() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scfxm1"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_agg2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/agg2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_finnis() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/finnis"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_dfl001() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/dfl001"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_pilot87() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/pilot87"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sctap1() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sctap1"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_agg3() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/agg3"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_grow7() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/grow7"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scorpion() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scorpion"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_maros() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/maros"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_shell() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/shell"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_greenbeb() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/greenbeb"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sc50b() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sc50b"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_recipe() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/recipe"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sierra() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sierra"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scagr25() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scagr25"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_modszk1() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/modszk1"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_ship12l() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/ship12l"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_stair() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/stair"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_cycle() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/cycle"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sc105() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sc105"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_pilot_ja() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/pilot.ja"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_beaconfd() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/beaconfd"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_czprob() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/czprob"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_pilot_we() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/pilot.we"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_standgub() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/standgub"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_standmps() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/standmps"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scsd8() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scsd8"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_woodw() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/woodw"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scsd6() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scsd6"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scsd1() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scsd1"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_share2b() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/share2b"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_gfrd_pnc() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/gfrd-pnc"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_bnl2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/bnl2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_stocfor2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/stocfor2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_nesm() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/nesm"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_share1b() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/share1b"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_ship04s() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/ship04s"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_grow15() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/grow15"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_maros_r7() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/maros-r7"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_blend() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/blend"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_lotfi() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/lotfi"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_standata() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/standata"
    ))?);
    Ok(())
  }

  ////#[test]
  ////fn test_parse_d6cube() -> Result<()> {
  //  insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
  //    "../tests/data/netlib/d6cube"
  //  ))?);
  //  Ok(())
  ////}

  #[test]
  fn test_parse_degen3() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/degen3"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_capri() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/capri"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_grow22() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/grow22"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_etamacro() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/etamacro"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_ship08l() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/ship08l"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_afiro() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/afiro"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_degen2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/degen2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_boeing2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/boeing2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_fit1d() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/fit1d"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scfxm2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scfxm2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sctap3() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sctap3"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_fit1p() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/fit1p"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_pilot() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/pilot"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_fit2d() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/fit2d"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sctap2() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sctap2"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scfxm3() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scfxm3"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_brandy() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/brandy"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_greenbea() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/greenbea"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_tuff() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/tuff"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_sc50a() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/sc50a"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_vtp_base() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/vtp.base"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_pilotnov() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/pilotnov"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_ship12s() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/ship12s"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_seba() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/seba"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_fffff800() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/fffff800"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_israel() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/israel"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_perold() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/perold"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_pilot4() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/pilot4"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_scagr7() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/scagr7"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_bandm() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/bandm"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_bnl1() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/bnl1"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_stocfor1() -> Result<()> {
    insta::assert_yaml_snapshot!(parse::<f32>(include_str!(
      "../tests/data/netlib/stocfor1"
    ))?);
    Ok(())
  }
}
