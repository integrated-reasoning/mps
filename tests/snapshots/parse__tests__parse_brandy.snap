---
source: tests/parse.rs
expression: "parse::<f32>(include_str!(\"../tests/data/netlib/brandy\"))?"
---
name: BRANDY
rows:
  - row_type: Nr
    row_name: 10000A
  - row_type: Eq
    row_name: 10001A
  - row_type: Eq
    row_name: 10002A
  - row_type: Eq
    row_name: 10003A
  - row_type: Eq
    row_name: 10004A
  - row_type: Leq
    row_name: 10006A
  - row_type: Leq
    row_name: 10007A
  - row_type: Eq
    row_name: 10008A
  - row_type: Leq
    row_name: 10009A
  - row_type: Leq
    row_name: 10010A
  - row_type: Leq
    row_name: 10011A
  - row_type: Leq
    row_name: 10012A
  - row_type: Leq
    row_name: 10013A
  - row_type: Eq
    row_name: 10014A
  - row_type: Leq
    row_name: 10015A
  - row_type: Leq
    row_name: 10016A
  - row_type: Leq
    row_name: 10017A
  - row_type: Leq
    row_name: 10018A
  - row_type: Leq
    row_name: 10019A
  - row_type: Leq
    row_name: 10020A
  - row_type: Leq
    row_name: 10021A
  - row_type: Leq
    row_name: 10022A
  - row_type: Leq
    row_name: 10023A
  - row_type: Leq
    row_name: 10024A
  - row_type: Eq
    row_name: 10025A
  - row_type: Eq
    row_name: 10026A
  - row_type: Eq
    row_name: 10027A
  - row_type: Eq
    row_name: 10028A
  - row_type: Eq
    row_name: 10029A
  - row_type: Eq
    row_name: 10030A
  - row_type: Eq
    row_name: 10031A
  - row_type: Eq
    row_name: 10032A
  - row_type: Eq
    row_name: 10033A
  - row_type: Eq
    row_name: 10034A
  - row_type: Eq
    row_name: 10035A
  - row_type: Eq
    row_name: 10036A
  - row_type: Eq
    row_name: 10037A
  - row_type: Eq
    row_name: 10038A
  - row_type: Eq
    row_name: 10039A
  - row_type: Leq
    row_name: 10040A
  - row_type: Eq
    row_name: 10041A
  - row_type: Eq
    row_name: 10042A
  - row_type: Eq
    row_name: 10043A
  - row_type: Eq
    row_name: 10044A
  - row_type: Eq
    row_name: 10045A
  - row_type: Eq
    row_name: 10046A
  - row_type: Eq
    row_name: 10047A
  - row_type: Eq
    row_name: 10048A
  - row_type: Eq
    row_name: 10049A
  - row_type: Eq
    row_name: 10050A
  - row_type: Eq
    row_name: 10051A
  - row_type: Eq
    row_name: 10052A
  - row_type: Eq
    row_name: 10053A
  - row_type: Eq
    row_name: 10054A
  - row_type: Eq
    row_name: 10055A
  - row_type: Eq
    row_name: 10056A
  - row_type: Eq
    row_name: 10057A
  - row_type: Eq
    row_name: 10058A
  - row_type: Leq
    row_name: 10059A
  - row_type: Leq
    row_name: 10060A
  - row_type: Eq
    row_name: 10061A
  - row_type: Eq
    row_name: 10062A
  - row_type: Eq
    row_name: 10063A
  - row_type: Eq
    row_name: 10064A
  - row_type: Eq
    row_name: 10065A
  - row_type: Eq
    row_name: 10066A
  - row_type: Eq
    row_name: 10067A
  - row_type: Eq
    row_name: 10068A
  - row_type: Eq
    row_name: 10069A
  - row_type: Eq
    row_name: 10070A
  - row_type: Eq
    row_name: 10071A
  - row_type: Eq
    row_name: 10072A
  - row_type: Eq
    row_name: 10073A
  - row_type: Eq
    row_name: 10074A
  - row_type: Eq
    row_name: 10075A
  - row_type: Eq
    row_name: 10076A
  - row_type: Eq
    row_name: 10077A
  - row_type: Eq
    row_name: 10078A
  - row_type: Eq
    row_name: 10079A
  - row_type: Eq
    row_name: 10080A
  - row_type: Eq
    row_name: 10081A
  - row_type: Eq
    row_name: 10082A
  - row_type: Eq
    row_name: 10083A
  - row_type: Eq
    row_name: 10084A
  - row_type: Eq
    row_name: 10085A
  - row_type: Eq
    row_name: 10086A
  - row_type: Eq
    row_name: 10087A
  - row_type: Eq
    row_name: 10088A
  - row_type: Leq
    row_name: 10089A
  - row_type: Eq
    row_name: 10090A
  - row_type: Eq
    row_name: 10091A
  - row_type: Eq
    row_name: 10092A
  - row_type: Eq
    row_name: 10093A
  - row_type: Eq
    row_name: 10094A
  - row_type: Eq
    row_name: 10095A
  - row_type: Eq
    row_name: 10096A
  - row_type: Eq
    row_name: 10097A
  - row_type: Eq
    row_name: 10098A
  - row_type: Eq
    row_name: 10099A
  - row_type: Eq
    row_name: 10100A
  - row_type: Eq
    row_name: 10101A
  - row_type: Eq
    row_name: 10102A
  - row_type: Eq
    row_name: 10103A
  - row_type: Eq
    row_name: 10104A
  - row_type: Eq
    row_name: 10105A
  - row_type: Eq
    row_name: 10106A
  - row_type: Eq
    row_name: 10107A
  - row_type: Eq
    row_name: 10108A
  - row_type: Eq
    row_name: 10109A
  - row_type: Eq
    row_name: 10110A
  - row_type: Eq
    row_name: 10111A
  - row_type: Eq
    row_name: 10112A
  - row_type: Eq
    row_name: 10113A
  - row_type: Eq
    row_name: 10114A
  - row_type: Eq
    row_name: 10115A
  - row_type: Eq
    row_name: 10116A
  - row_type: Eq
    row_name: 10117A
  - row_type: Eq
    row_name: 10118A
  - row_type: Eq
    row_name: 10119A
  - row_type: Eq
    row_name: 10120A
  - row_type: Eq
    row_name: 10121A
  - row_type: Leq
    row_name: 10122A
  - row_type: Eq
    row_name: 10123A
  - row_type: Eq
    row_name: 10124A
  - row_type: Leq
    row_name: 10125A
  - row_type: Eq
    row_name: 10126A
  - row_type: Leq
    row_name: 10127A
  - row_type: Eq
    row_name: 10128A
  - row_type: Eq
    row_name: 10129A
  - row_type: Leq
    row_name: 10130A
  - row_type: Leq
    row_name: 10131A
  - row_type: Leq
    row_name: 10132A
  - row_type: Eq
    row_name: 10133A
  - row_type: Eq
    row_name: 10134A
  - row_type: Eq
    row_name: 10135A
  - row_type: Leq
    row_name: 10136A
  - row_type: Leq
    row_name: 10137A
  - row_type: Eq
    row_name: 10138A
  - row_type: Leq
    row_name: 10139A
  - row_type: Eq
    row_name: 10140A
  - row_type: Eq
    row_name: 10141A
  - row_type: Eq
    row_name: 10142A
  - row_type: Eq
    row_name: 10143A
  - row_type: Eq
    row_name: 10144A
  - row_type: Eq
    row_name: 10145A
  - row_type: Eq
    row_name: 10146A
  - row_type: Eq
    row_name: 10147A
  - row_type: Eq
    row_name: 10148A
  - row_type: Eq
    row_name: 10149A
  - row_type: Eq
    row_name: 10150A
  - row_type: Eq
    row_name: 10151A
  - row_type: Eq
    row_name: 10152A
  - row_type: Eq
    row_name: 10153A
  - row_type: Eq
    row_name: 10154A
  - row_type: Eq
    row_name: 10155A
  - row_type: Eq
    row_name: 10156A
  - row_type: Eq
    row_name: 10157A
  - row_type: Eq
    row_name: 10158A
  - row_type: Eq
    row_name: 10159A
  - row_type: Leq
    row_name: 10160A
  - row_type: Eq
    row_name: 10161A
  - row_type: Leq
    row_name: 10162A
  - row_type: Leq
    row_name: 10163A
  - row_type: Leq
    row_name: 10164A
  - row_type: Eq
    row_name: 10165A
  - row_type: Eq
    row_name: 10166A
  - row_type: Leq
    row_name: 10167A
  - row_type: Leq
    row_name: 10168A
  - row_type: Leq
    row_name: 10169A
  - row_type: Eq
    row_name: 10170A
  - row_type: Eq
    row_name: 10171A
  - row_type: Eq
    row_name: 10172A
  - row_type: Eq
    row_name: 10173A
  - row_type: Eq
    row_name: 10174A
  - row_type: Eq
    row_name: 10175A
  - row_type: Eq
    row_name: 10176A
  - row_type: Eq
    row_name: 10177A
  - row_type: Leq
    row_name: 10178A
  - row_type: Leq
    row_name: 10179A
  - row_type: Leq
    row_name: 10180A
  - row_type: Leq
    row_name: 10181A
  - row_type: Leq
    row_name: 10182A
  - row_type: Leq
    row_name: 10183A
  - row_type: Leq
    row_name: 10184A
  - row_type: Leq
    row_name: 10185A
  - row_type: Leq
    row_name: 10186A
  - row_type: Leq
    row_name: 10187A
  - row_type: Leq
    row_name: 10188A
  - row_type: Leq
    row_name: 10189A
  - row_type: Leq
    row_name: 10190A
  - row_type: Leq
    row_name: 10191A
  - row_type: Leq
    row_name: 10192A
  - row_type: Leq
    row_name: 10193A
  - row_type: Eq
    row_name: 10194A
  - row_type: Leq
    row_name: 10195A
  - row_type: Eq
    row_name: 10196A
  - row_type: Eq
    row_name: 10197A
  - row_type: Eq
    row_name: 10198A
  - row_type: Eq
    row_name: 10199A
  - row_type: Eq
    row_name: 10200A
  - row_type: Eq
    row_name: 10201A
  - row_type: Eq
    row_name: 10202A
  - row_type: Eq
    row_name: 10203A
  - row_type: Eq
    row_name: 10204A
  - row_type: Eq
    row_name: 10205A
  - row_type: Eq
    row_name: 10206A
  - row_type: Eq
    row_name: 10207A
  - row_type: Eq
    row_name: 10208A
  - row_type: Eq
    row_name: 10209A
  - row_type: Eq
    row_name: 10210A
  - row_type: Eq
    row_name: 10211A
  - row_type: Eq
    row_name: 10212A
  - row_type: Eq
    row_name: 10213A
  - row_type: Eq
    row_name: 10214A
  - row_type: Eq
    row_name: 10215A
  - row_type: Eq
    row_name: 10216A
  - row_type: Eq
    row_name: 10217A
  - row_type: Eq
    row_name: 10218A
  - row_type: Eq
    row_name: 10219A
  - row_type: Eq
    row_name: 10220A
  - row_type: Eq
    row_name: 10221A
columns:
  - name: "100001"
    first_pair:
      row_name: 10000A
      value: 1
    second_pair:
      row_name: 10056A
      value: -1
  - name: "100002"
    first_pair:
      row_name: 10000A
      value: 1
    second_pair:
      row_name: 10025A
      value: -1
  - name: "100100"
    first_pair:
      row_name: 10006A
      value: 1
    second_pair:
      row_name: 10008A
      value: 1
  - name: "100100"
    first_pair:
      row_name: 10025A
      value: 7
    second_pair:
      row_name: 10043A
      value: 0.003
  - name: "100100"
    first_pair:
      row_name: 10044A
      value: 0.067
    second_pair:
      row_name: 10045A
      value: 0.022
  - name: "100100"
    first_pair:
      row_name: 10056A
      value: 0.01
    second_pair:
      row_name: 10063A
      value: 0.22
  - name: "100100"
    first_pair:
      row_name: 10064A
      value: 0.1
    second_pair:
      row_name: 10065A
      value: 0.112
  - name: "100100"
    first_pair:
      row_name: 10066A
      value: 0.349
    second_pair:
      row_name: 10068A
      value: 0.125
rhs: ~
ranges: ~
bounds: ~

